### ISSUE TITLE
CustomTabsManagerActivity was created with a null state.

### ISSUE THREAD

**Issue by akshayshankarek** [aws-amplify/amplify-android#1101]
1. When I try to signout using Amplify.Auth.signout() , I get success callback but also an error saying "CustomTabsManagerActivity was created with a null state. This should never happen. Please file an issue with us on GitHub"
And after this if I try to sign in using Amplify.Auth.signInWithWebUI() , it gets signed in with previous user without prompting login screen.
2. Amplify framework core 1.6.8
3. Amplify.Auth.signout()

**Comment by raphkim** [comment]
Hi @akshayshankarek ,
Can you provide us with relevant code snippet where you are using the Auth API?

**Comment by akshayshankarek** [comment]
Hi @raphkim 
I have MainActivity which has two buttons , one for signIn and one for SignOut.
OnClick of signIn button I call below snippet : 
```
Amplify.Auth.signInWithWebUI(
    this,
    result -> Log.i("AuthQuickStart", result.toString()),
    error -> Log.e("AuthQuickStart", error.toString())
);
```
OnClick of signOut : 

```
Amplify.Auth.signOut(
    () -> Log.i("AuthQuickstart", "Signed out successfully"),
    error -> Log.e("AuthQuickstart", error.toString())
); 
```
I also incorporated the below snippet in AndroidManifest.xml

```
 <activity
      android:name="com.amazonaws.mobileconnectors.cognitoauth.activities.CustomTabsRedirectActivity">
      <intent-filter>
          <action android:name="android.intent.action.VIEW" />

          <category android:name="android.intent.category.DEFAULT" />
          <category android:name="android.intent.category.BROWSABLE" />

          <data android:scheme="app" />
      </intent-filter>
  </activity>
```
Also in MainActivity where I have signIn an Signout button I have implemented onActivityResult as below
```
@Override
protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    super.onActivityResult(requestCode, resultCode, data);

    if (requestCode == AWSCognitoAuthPlugin.WEB_UI_SIGN_IN_ACTIVITY_CODE) {
        Amplify.Auth.handleWebUISignInResponse(data);
    }
}
```
Dependencies used :
com.amplifyframework:core:1.6.8 
com.amplifyframework:aws-auth-cognito:1.6.8

I also have configured my amplifyconfiguration.json file with WebDomain , AppClientId, AppClientSecret, SignInedirectURI, SignOutRedirectURI.

The issue is that if I try to login with user , it prompts user login page and will be successfully logged in , but when I try to signOut , I get success callback as well as a log saying  "CustomTabsManagerActivity was created with a null state. This should never happen. Please file an issue with us on GitHub" .
After this, if I try to sign in , the previous user gets signed in without prompting the login page. 
But when I tried to just open and close chrome app and then try to re-sign in , it prompts the user login page of hosted UI.

**Comment by richardmcclellan** [comment]
Hi @akshayshankarek, I setup a quick sample app with your setup and I am able to reproduce this bug.  Thanks for reporting it!

That said, I am _not_ able to reproduce this bug with [my hosted UI sample app](https://github.com/richardmcclellan/amplify-android-samples/tree/main/auth-hostedui).  In my sample, I have a `SignInActivity`, with a "Sign In" button, which launches a `MainActivity` on successful sign in.  Then, the `MainActivity` has a "Sign Out" button, which calls `finish()` on itself on successful sign out.

I think the root cause has something to do with the fact that you are trying to call sign in and sign out from the same `Activity`.  I'd suggest an app more similar to the sample that I described as a workaround until we can address the root cause.

**Comment by changxu0306** [comment]
> Hi @akshayshankarek, I setup a quick sample app with your setup and I am able to reproduce this bug. Thanks for reporting it!
> 
> That said, I am _not_ able to reproduce this bug with [my hosted UI sample app](https://github.com/richardmcclellan/amplify-android-samples/tree/main/auth-hostedui). In my sample, I have a `SignInActivity`, with a "Sign In" button, which launches a `MainActivity` on successful sign in. Then, the `MainActivity` has a "Sign Out" button, which calls `finish()` on itself on successful sign out.
> 
> I think the root cause has something to do with the fact that you are trying to call sign in and sign out from the same `Activity`. I'd suggest an app more similar to the sample that I described as a workaround until we can address the root cause.

I'm able to reproduce this bug in my sample app. But there, sign in and sign out are from two different activities.

**Comment by raphkim** [comment]
This issue is now resolved with [v1.17.8](https://github.com/aws-amplify/amplify-android/releases/tag/release_v1.17.8). Please follow the [updated documentation](https://docs.amplify.aws/lib/auth/signin_web_ui/q/platform/android#update-androidmanifestxml) to apply this fix :)
